src/odemis/driver/orsay.py

```diff
@@ -1530,12 +1556,15 @@ def __init__(self, va, parameter, attr_name="Actual", conversion=None, factor=No
                 else:
                     upperbound = p.Max
                 if upperbound is not None:  # if an upperbound is defined in the server
-                    new_range[1][i] = self._parameter_to_VA_value(upperbound)  # copy it to the va
+                    if self._va_is_tuple:
+                        new_range[1][i] = self._parameter_to_VA_value(upperbound)  # copy it to the va
+                    else:
+                        new_range[1] = self._parameter_to_VA_value(upperbound)  # copy it to the va
 
             if self._va_is_tuple:
-                new_range = (new_range[0][0], new_range[1][0])
+                new_range = (new_range[0], new_range[1])
             else:
-                new_range = (tuple(new_range[0]), tuple(new_range[1]))
+                new_range = tuple(new_range)
```

Comment:
In my opinion the rest of this method/class isn't particularly simple actually. Because of the in-explicit distinction between ranges with 1 and 2 limits  ( like [-1, +1] vs [ [-1, -2], [1, 2] ] ) it took my a while to figure out how the exact cause/fix for all the failing test cases.
Even though my solution means more lines of code, I think that the readability increases. The explicit distinction between both type of ranges helps me to understand what is happening. It shows that the same thing is applied for both type of ranges but acknowledges the differences explicitly.

Both implementations work so I in the end it doesn't matter much though.

Reply:
I do agree that the original version is bit hard to understand. Let's go with your implementation then.